_posts/articles/2021-02-14-setting-up-techdocs-on-backstage.md

```diff
@@ -0,0 +1,210 @@
+
+In this post I will take you through the following
+
+- Learn basic concepts and structure of backstage and techdocs
+- Install Backstage
+- Setup tech docs
+- Creating and publishing documentation.
+- All on local machine
+
+## **Basics concepts and structures of backstage and techdocs**
+
+The focus of this article is techdocs so i will go through the other main features at a high level.
+
+- Service catalog. It keeps track of ownership and metadata for all the software in your ecosystem. it does this by putting metadata in YAML files stored together with your code. You process these files and then you can visualise the catalog in backstage. This will enable your teams to manage and maintain the software they own. it also makes the software discoverable.
+- Software templates - This feature allows you to create templates or skeletons of code. These templates are published to github.
+- Plugins - Allow you to integrate third part tools or any kind of infrastructure into backstage. They are open source and you can view a list [here](https://backstage.io/plugins)
+- Techdocs- yay, finally, why we are here. Techdocs is Spotify's homegrown docs like code solution. it allows the user to store documentation to near code thus allowing it to be easily discovered.
+
+When you deploy backstage with Techdocs enabled you get a basic out of the box experience.
+
+At its core it is MKDocs plugin and other MkDocs plugins and Python Markdown extensions which allows it to standardize the configuration of MkDocs used for TechDocs.
+
+You can see the code [here](https://github.com/backstage/mkdocs-techdocs-core)
+
+The other moving parts are
+
+- The techdocs container which can be found on Docker-hub, It builds static content through MKDocs
+- The Techdocs backend plugin This is the backend part of the techdocs plugin.
+- The Techdocs CLI, Command line tool for managing techdocs sites in backstage
+- The Techdocs reader, it fetch's remotes pages , run transformers against them and renders them in a shadow DOM
+- Transformers API is a function that takes in parameters from the reader component and returns a function which gets passed the dom of the fetched page.
+
+Architecture
+
+![Architecture diagram](https://backstage.io/docs/assets/techdocs/architecture-basic.drawio.svg)
+
+When you open a techdocs sites a request is made the techdocs reader calls the techdocs-backend with the entity id and the path of the current page, the response contains the static content
+
+The static content contains HTML and CSS, javascript is removed for security reasons
+
+Transforms are then applied which modiy the generated static HTML files for a number of reasons like removing certain headers etc
+
+For the following instructions we will use the local file storage but it is better to use cloud storage like S3.
+
+## **Install Backstage**
+
+### Perquisites
+Mac running MacOS
+
+Node version 14.x
```

Comment:
These should be bullets. Like [this example](https://deploy-preview-167--docs-like-code.netlify.app/articles/setting-up-techdocs-on-backstage/).